oclc.py

```diff
@@ -507,119 +558,34 @@ def info_for_book_graph(self, subgraph, book_info):
                     elif len(isbn) != 13:
                         continue
                     if isbn:
-                        isbns.add(isbn)
+                        metadata.identifiers.append(IdentifierData(
+                            type = Identifier.ISBN, identifier = isbn
+                        ))
 
         # Consolidate subjects and apply a blacklist.
-        tags = set()
-        for tag in subjects.get(Subject.TAG, []):
-            fixed = self._fix_tag(tag)
-            if fixed == self.UNUSABLE_RECORD:
-                return None
-            elif fixed:
-                tags.add(fixed)
-        if tags:
-            subjects[Subject.TAG] = tags
-        elif Subject.TAG in subjects:
-            del subjects[Subject.TAG]
-
-        # Something interesting has to come out of this
-        # work--something we couldn't get from another source--or
-        # there's no point.
-        if not isbns and not descriptions and not subjects:
-            return None
-
-        publishers = self.internal_lookup(
-            subgraph, publisher_uris)
-        publisher_names = [
-            i['schema:name'] for i in publishers
-            if 'schema:name' in i]
-        publisher_names = list(ldq.values(
-            ldq.restrict_to_language(publisher_names, 'en')))
-
-        for n in publisher_names:
-            if (n in self.PUBLISHER_BLACKLIST
-                or 'Audio' in n or 'Video' in n or 'Tape' in n
-                or 'Comic' in n or 'Music' in n):
-                # This book is from a publisher that will probably not
-                # give us metadata we can use.
-                return None
-
-        # Project Gutenberg texts don't have ISBNs, so if there's an
-        # ISBN on there, it's probably wrong. Unless someone stuck a
-        # description on there, there's no point in discussing
-        # OCLC+LD's view of a Project Gutenberg work.
-        if 'Project Gutenberg' in publisher_names and not descriptions:
+        fixed = [self._fix_tag(tag) for tag in subjects.get(Subject.TAG, [])]
+        fixed_tags = [tag for tag in fixed if not None ]
+        subjects[Subject.TAG] = fixed_tags
+
+        for subject_type, subject_identifiers in subjects.items():
+            for subject_identifier in subject_identifiers:
+                metadata.subjects.append(SubjectData(
+                    type=subject_type, identifier=subject_identifier
+                ))
+
+        if (not metadata.links and not metadata.identifiers and
```

Comment:
I think finding some good contributor information (which you check for below) would be another reason to not return None.